### PR TITLE
Clearer documentation for the ATTACK_AND_VALIDATE macro

### PR DESCRIPTION
**Pull request by stevecotton**
@newfrenchy83 please would you review whether this PR is a good change?

### CODE REVIEW

**Comment by newfrenchy83** [comment]
the documentation is more complete for me

File: data/test/macros/wml_unit_test_macros.cfg [hunk 1]
```diff
@@ -76,9 +76,13 @@
     [/if]
 #enddef
 
-# have one adjacent unit attack another adjacent unit
-# each unit is given 1000 hp and their attack is made to 1x100, with a 100% chance of hitting
 #define ATTACK_AND_VALIDATE DAMAGE1
+    # Have one adjacent unit attack another adjacent unit.
+    # Each unit is given 1000 hp and their attack is made to 1x100, with a 100% chance of hitting.
```
Comment (ProditorMagnus) [inline data/test/macros/wml_unit_test_macros.cfg:81]: 100-1

**Comment by stevecotton** [comment]
Corrected the "1x100", and indented the other docs in this file to match the style in data/core/macros/

**Review by Pentarctagon — APPROVED** [review]

**Comment by soliton-** [comment]
I do not think the habit of putting documentation inside the macro is a good one. Those comments are then part of the macro content and may break certain macro use.

I know it's done elsewhere because the macro reference generation uses it but this is not relevant here as far as I can tell.

**Comment by stevecotton** [comment]
> may break certain macro use

How? I'm okay with taking out that part of the change, but I'm interested in how it might break.

**Comment by soliton-** [comment]
Any use that puts the macro content in quotes or such so that the comments are not comments anymore. So for example: `"{MACRO}"`

**Comment by stevecotton** [comment]
Changed back to having documentation before macros instead of in them. Planning to merge tomorrow, as that seems to fit with all the comments.

**Pull request merged**